network/message/authorization.go

```diff
@@ -0,0 +1,364 @@
+		},
+	}
+	authorizationConfigs[EntityResponse] = MsgAuthConfig{
+		Name: EntityResponse,
+		Type: func() interface{} {
+			return new(messages.EntityResponse)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.ProvideApprovalsByChunk:  {flow.RoleVerification},
+			channels.ProvideReceiptsByBlockID: {flow.RoleExecution},
+		},
+	}
+
+	// testing
+	authorizationConfigs[TestMessage] = MsgAuthConfig{
+		Name: TestMessage,
+		Type: func() interface{} {
+			return new(message.TestMessage)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.TestNetworkChannel: flow.Roles(),
+			channels.TestMetricsChannel: flow.Roles(),
+		},
+	}
+
+	// DKG
+	authorizationConfigs[DKGMessage] = MsgAuthConfig{
+		Name: DKGMessage,
+		Type: func() interface{} {
+			return new(messages.DKGMessage)
+		},
+		Config: map[channels.Channel]flow.RoleList{
+			channels.DKGCommittee: {flow.RoleConsensus},
+		},
+	}
+}
+
+// GetMessageAuthConfig checks the underlying type and returns the correct
+// message auth Config.
+// Expected error returns during normal operations:
+//  * ErrUnknownMsgType : if underlying type of v does  not match any of the known message types
+func GetMessageAuthConfig(v interface{}) (MsgAuthConfig, error) {
+	switch v.(type) {
+	// consensus
+	case *messages.BlockProposal:
+		return authorizationConfigs[BlockProposal], nil
+	case *messages.BlockVote:
+		return authorizationConfigs[BlockVote], nil
+
+	// protocol state sync
+	case *messages.SyncRequest:
+		return authorizationConfigs[SyncRequest], nil
+	case *messages.SyncResponse:
+		return authorizationConfigs[SyncResponse], nil
+	case *messages.RangeRequest:
+		return authorizationConfigs[RangeRequest], nil
+	case *messages.BatchRequest:
+		return authorizationConfigs[BatchRequest], nil
+	case *messages.BlockResponse:
+		return authorizationConfigs[BlockResponse], nil
+
+	// cluster consensus
+	case *messages.ClusterBlockProposal:
+		return authorizationConfigs[ClusterBlockProposal], nil
+	case *messages.ClusterBlockVote:
+		return authorizationConfigs[ClusterBlockVote], nil
+	case *messages.ClusterBlockResponse:
+		return authorizationConfigs[ClusterBlockResponse], nil
+
+	// collections, guarantees & transactions
+	case *flow.CollectionGuarantee:
+		return authorizationConfigs[CollectionGuarantee], nil
+	case *flow.TransactionBody:
+		return authorizationConfigs[TransactionBody], nil
+	case *flow.Transaction:
+		return authorizationConfigs[Transaction], nil
+
+	// core messages for execution & verification
+	case *flow.ExecutionReceipt:
+		return authorizationConfigs[ExecutionReceipt], nil
+	case *flow.ResultApproval:
+		return authorizationConfigs[ResultApproval], nil
+
+	// execution state synchronization
+	case *messages.ExecutionStateSyncRequest:
+		return authorizationConfigs[ExecutionStateSyncRequest], nil
+	case *messages.ExecutionStateDelta:
+		return authorizationConfigs[ExecutionStateDelta], nil
```

Comment:
similarly here, there is no configuration allowing any role to send such messages. The returned `MsgAuthConfig` will be the zero value, which I am worried that the processing code will not properly be able to handle.

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/a0e47046e1f386435c4ce4a6bc16ff1ee2ce32f9
https://github.com/onflow/flow-go/pull/2648/commits/6d62de313106e6cf81b067b0f8ad756786306cf7